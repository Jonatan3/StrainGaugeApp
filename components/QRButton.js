import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { TouchableOpacity } from 'react-native'
import { Scan_qr } from '../resources/svgs'
import colors from '../resources/colors'

export default function QRCodeButton({ onPressHandler }) {
  return (
    <View style={{ flex: 1, alignSelf: 'center' }}>
      <TouchableOpacity onPress={() => onPressHandler()} >
        <View style={styles.button}>
          <Text style={{ color: 'white', padding: 8, paddingLeft: 16 }}>
            Scan QR
          </Text>
          <View style={{ marginRight: 16 }}>
            <Scan_qr />
          </View>
        </View>
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.qrButton, 
    flexDirection: 'row', 
    borderRadius: 20, 
    borderBottomWidth: 2, 
    borderRightWidth: 1, 
    borderLeftWidth: 1, 
    borderColor: colors.qrBorder,
    alignItems: 'center',
    justifyContent: 'center'
  }
})