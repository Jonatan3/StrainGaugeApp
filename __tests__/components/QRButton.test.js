import React from 'react';
import renderer from 'react-test-renderer';
import QRButton from '../../components/QRButton'

test('renders correctly', () => {
    const tree = renderer.create(<QRButton />).toJSON()
    expect(tree).toMatchSnapshot()
})