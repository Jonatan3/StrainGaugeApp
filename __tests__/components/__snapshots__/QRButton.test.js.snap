// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2DA39D",
          "borderBottomWidth": 2,
          "borderColor": "#00817C",
          "borderLeftWidth": 1,
          "borderRadius": 20,
          "borderRightWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "padding": 8,
            "paddingLeft": 16,
          }
        }
      >
        Scan QR
      </Text>
      <View
        style={
          Object {
            "marginRight": 16,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="18"
          bbWidth="18"
          fill="none"
          focusable={false}
          height="18"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 18,
                "width": 18,
              },
            ]
          }
          vbHeight={18}
          vbWidth={18}
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M6.75 4.04951V6.74951H4.05V4.04951H6.75ZM8.1 2.69951H2.7V8.09951H8.1V2.69951ZM6.75 11.2495V13.9495H4.05V11.2495H6.75ZM8.1 9.89951H2.7V15.2995H8.1V9.89951ZM13.95 4.04951V6.74951H11.25V4.04951H13.95ZM15.3 2.69951H9.9V8.09951H15.3V2.69951ZM9.9 9.89951H11.25V11.2495H9.9V9.89951ZM11.25 11.2495H12.6V12.5995H11.25V11.2495ZM12.6 9.89951H13.95V11.2495H12.6V9.89951ZM9.9 12.5995H11.25V13.9495H9.9V12.5995ZM11.25 13.9495H12.6V15.2995H11.25V13.9495ZM12.6 12.5995H13.95V13.9495H12.6V12.5995ZM13.95 11.2495H15.3V12.5995H13.95V11.2495ZM13.95 13.9495H15.3V15.2995H13.95V13.9495ZM17.1 4.49951C16.8613 4.49951 16.6324 4.40469 16.4636 4.23591C16.2948 4.06712 16.2 3.83821 16.2 3.59951V1.79951H14.4C14.1613 1.79951 13.9324 1.70469 13.7636 1.53591C13.5948 1.36713 13.5 1.13821 13.5 0.899512C13.5 0.660817 13.5948 0.431898 13.7636 0.263116C13.9324 0.0943328 14.1613 -0.000488281 14.4 -0.000488281H17.1C17.3387 -0.000488281 17.5676 0.0943328 17.7364 0.263116C17.9052 0.431898 18 0.660817 18 0.899512V3.59951C18 3.83821 17.9052 4.06712 17.7364 4.23591C17.5676 4.40469 17.3387 4.49951 17.1 4.49951ZM18 17.0995V14.3995C18 14.1608 17.9052 13.9319 17.7364 13.7631C17.5676 13.5943 17.3387 13.4995 17.1 13.4995C16.8613 13.4995 16.6324 13.5943 16.4636 13.7631C16.2948 13.9319 16.2 14.1608 16.2 14.3995V16.1995H14.4C14.1613 16.1995 13.9324 16.2943 13.7636 16.4631C13.5948 16.6319 13.5 16.8608 13.5 17.0995C13.5 17.3382 13.5948 17.5671 13.7636 17.7359C13.9324 17.9047 14.1613 17.9995 14.4 17.9995H17.1C17.3387 17.9995 17.5676 17.9047 17.7364 17.7359C17.9052 17.5671 18 17.3382 18 17.0995ZM0.9 17.9995H3.6C3.83869 17.9995 4.06761 17.9047 4.2364 17.7359C4.40518 17.5671 4.5 17.3382 4.5 17.0995C4.5 16.8608 4.40518 16.6319 4.2364 16.4631C4.06761 16.2943 3.83869 16.1995 3.6 16.1995H1.8V14.3995C1.8 14.1608 1.70518 13.9319 1.5364 13.7631C1.36761 13.5943 1.13869 13.4995 0.9 13.4995C0.661305 13.4995 0.432387 13.5943 0.263604 13.7631C0.0948211 13.9319 0 14.1608 0 14.3995V17.0995C0 17.3382 0.0948211 17.5671 0.263604 17.7359C0.432387 17.9047 0.661305 17.9995 0.9 17.9995ZM0 0.899512V3.59951C0 3.83821 0.0948211 4.06712 0.263604 4.23591C0.432387 4.40469 0.661305 4.49951 0.9 4.49951C1.13869 4.49951 1.36761 4.40469 1.5364 4.23591C1.70518 4.06712 1.8 3.83821 1.8 3.59951V1.79951H3.6C3.83869 1.79951 4.06761 1.70469 4.2364 1.53591C4.40518 1.36713 4.5 1.13821 4.5 0.899512C4.5 0.660817 4.40518 0.431898 4.2364 0.263116C4.06761 0.0943328 3.83869 -0.000488281 3.6 -0.000488281H0.9C0.661305 -0.000488281 0.432387 0.0943328 0.263604 0.263116C0.0948211 0.431898 0 0.660817 0 0.899512V0.899512Z"
              fill={4294967295}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
